kofu/src/test/kotlin/org/springframework/fu/kofu/jdbc/JdbcDslTests.kt

```diff
@@ -1,17 +1,25 @@
 package org.springframework.fu.kofu.jdbc
 
 import org.junit.Assert
+import org.junit.jupiter.api.Assertions.*
 import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.getBean
+import org.springframework.boot.autoconfigure.jdbc.DataSourceType
 import org.springframework.fu.kofu.application
 import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.testcontainers.containers.GenericContainer
+import java.sql.PreparedStatement
 
 class JdbcDslTests {
 
     @Test
     fun `enable jdbc`() {
         val app = application {
-            jdbc()
+            jdbc {
```

Comment:
I am tempted to add a mandatory parameter `DataSourceType` to JDBC, otherwise I think it could be confusing for users and the embedded versus others choice seems pretty critical. Something like that:
```
jdbc(DataSourceType.Embedded) {
  ...
}
```

Reply:
Before going further implementing this one we should probably discuss the proposition which has been made in #325 and which is really interesting (externalizing the `datasource` out of `JdbcDsl`)

Reply:
After a deeper look, I think I would be in favor to not externalizing `datasource`. The mapping is 1:1, the use case mentioned ("This solution could propose to use both system in same time of course (why not? Could be good during migration between both systems") is really niche, and that would be not consistent with what we do on R2DBC side. JDBC and datasources are one of the most complex piece of Boot configuration, I tend to think at least for a start it would be better to simplify how we present it to the user (so to keep what we have + the mandatory `DataSourceType` parameter).

Reply:
I;m ok with that.
I like the datasource as external confguration of the JDBC one but I understand the arguments to keep it in the DSL

The only thing I have reservatiions on is the mandatory arguments,  having a default parameters seems like a godd idea, we just defined `Generic` one as default that let a empty `jdbc {}` fail but we could change it. Or maybe have something to detect wich one we should use.

Bt in order to advance I'm for adding a mandatory parameter and create an issue about user experience on this subject